README.md
# Flutter Boom Menu

## Usage

The BoomMenu widget is built to be placed in the `Scaffold.floatingActionButton` argument, replacing the `FloatingActionButton` widget.
It's not possible to set its position with the `Scaffold.floatingActionButtonLocation` argument, but it's possible to set right/bottom margin with the `marginRight` and `marginBottom` arguments (default to 16) to place the button anywhere in the screen.
Using the `Scaffold.bottomNavigationBar` the floating button will be always placed above the bar, so the `BottomAppBar.hasNotch` should be always `false`.

**Title**

Every child button can have a `Icon`,`Title`, `SubTitle` which can be customized providing by yourself. If the `Title` parameter is not provided the title will be not rendered.

The package will handle the animation by itself.

![alt text](https://github.com/Mohanraj153/flutter_boom_menu/blob/master/screenshot/boom-menu-video.gif)

**Example Usage ( complete with all params ):**

```dart
Widget build(BuildContext context) {
    return Scaffold(
        floatingActionButton: BoomMenu(
      animatedIcon: AnimatedIcons.menu_close,
      animatedIconTheme: IconThemeData(size: 22.0),
      //child: Icon(Icons.add),
      onOpen: () => print('OPENING DIAL'),
      onClose: () => print('DIAL CLOSED'),
      scrollVisible: scrollVisible,
      overlayColor: Colors.black,
      overlayOpacity: 0.7,
      children: [
        MenuItem(
          child: Icon(Icons.accessibility, color: Colors.black),
          title: "Profiles",
          titleColor: Colors.white,
          subtitle: "You Can View the Noel Profile",
          subTitleColor: Colors.white,
          backgroundColor: Colors.deepOrange,
          onTap: () => print('FIRST CHILD'),
        ),
        MenuItem(
          child: Icon(Icons.brush, color: Colors.black),
          title: "Profiles",
          titleColor: Colors.white,
          subtitle: "You Can View the Noel Profile",
          subTitleColor: Colors.white,
          backgroundColor: Colors.green,
          onTap: () => print('SECOND CHILD'),
        ),
        MenuItem(
          child: Icon(Icons.keyboard_voice, color: Colors.black),
          title: "Profile",
          titleColor: Colors.white,
          subtitle: "You Can View the Noel Profile",
          subTitleColor: Colors.white,
          backgroundColor: Colors.blue,
          onTap: () => print('THIRD CHILD'),
        ),
        MenuItem(
          child: Icon(Icons.ac_unit, color: Colors.black),
          title: "Profiles",
          titleColor: Colors.white,
          subtitle: "You Can View the Noel Profile",
          subTitleColor: Colors.white,
          backgroundColor: Colors.blue,
          onTap: () => print('FOURTH CHILD'),
        )
      ],
    ),
    );
}
```
## Issues & Feedback

Please file an [issue](https://github.com/Mohanraj153/flutter_boom_menu/issues) to send feedback or report a bug. Thank you!

## Contributing

Every pull request is welcome.

CHANGELOG.md
## 1.0.2

* Now user can handle the body of scaffold widget.
* Improved Performance

## 1.0.1

* Now user can handle the body of scaffold widget.

## 1.0.0

* Now user can set alignment and padding to FAB.
* A better way of handling vertical overflows any MenuItem number is now supported.

## 0.0.2

* Fix Some Bugs.

## 0.0.2

* Fix Some Bugs.

## 0.0.1

* Initial release.


example/lib/main.dart
/// Created by Mohanraj G on 12-06-2020.
/// Email : mohanraj153@gmail.com

import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

import 'package:flutter_boom_menu/flutter_boom_menu.dart';

void main() {
  runApp(MaterialApp(home: MyApp(), title: 'Boom Menu Example'));
}

class MyApp extends StatefulWidget {
  @override
  MyAppState createState() => MyAppState();
}

class MyAppState extends State<MyApp> with TickerProviderStateMixin {
  ScrollController scrollController;
  bool scrollVisible = true;

  @override
  void initState() {
    super.initState();

    scrollController = ScrollController()
      ..addListener(() {
        setDialVisible(scrollController.position.userScrollDirection == ScrollDirection.forward);
      });
  }

  void setDialVisible(bool value) {
    setState(() {
      scrollVisible = value;
    });
  }

  Widget buildBody() {
    return ListView.builder(
      controller: scrollController,
      itemCount: 30,
      itemBuilder: (ctx, i) => ListTile(title: Text('Item $i')),
    );
  }

  BoomMenu buildBoomMenu() {
    return BoomMenu(
      animatedIcon: AnimatedIcons.menu_close,
      animatedIconTheme: IconThemeData(size: 22.0),
      //child: Icon(Icons.add),
      onOpen: () => print('OPENING DIAL'),
      onClose: () => print('DIAL CLOSED'),
      scrollVisible: scrollVisible,
      overlayColor: Colors.black,
      overlayOpacity: 0.7,
      children: [
        MenuItem(
//          child: Icon(Icons.accessibility, color: Colors.black, size: 40,),
          child: Image.asset('assets/logout_icon.png', color: Colors.grey[850]),
          title: "Logout",
          titleColor: Colors.grey[850],
          subtitle: "Lorem ipsum dolor sit amet, consectetur",
          subTitleColor: Colors.grey[850],
          backgroundColor: Colors.grey[50],
          onTap: () => print('THIRD CHILD'),
        ),
        MenuItem(
          child: Image.asset('assets/schemes_icon.png', color: Colors.white),
          title: "List",
          titleColor: Colors.white,
          subtitle: "Lorem ipsum dolor sit amet, consectetur",
          subTitleColor: Colors.white,
          backgroundColor: Colors.pinkAccent,
          onTap: () => print('FOURTH CHILD'),
        ),
        MenuItem(
          child: Image.asset('assets/customers_icon.png', color: Colors.grey[850]),
          title: "Team",
          titleColor: Colors.grey[850],
          subtitle: "Lorem ipsum dolor sit amet, consectetur",
          subTitleColor: Colors.grey[850],
          backgroundColor: Colors.grey[50],
          onTap: () => print('THIRD CHILD'),
        ),
        MenuItem(
          child: Image.asset('assets/profile_icon.png', color: Colors.white),
          title: "Profile",
          titleColor: Colors.white,
          subtitle: "Lorem ipsum dolor sit amet, consectetur",
          subTitleColor: Colors.white,
          backgroundColor: Colors.blue,
          onTap: () => print('FOURTH CHILD'),
        )
      ]
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('Boom Menu Example')),
      body: buildBody(),
      floatingActionButton: buildBoomMenu(),
    );
  }
}
